fix(JsonPipe): marks json pipe as not pure

sduquej opened the pull request:
Marked json pipe as not pure so that it runs all the time, reflecting changes on the object.

Closes #4821

IgorMinar commented:
please squash into a single commit and add a test.

Otherwise this looks the right change.

cc: @vsavkin

sduquej committed:
fix(JsonPipe): marks json pipe as not pure
Marked json pipe as not pure so that it runs all the time, reflecting changes on the object.

Closes #4821

vicb commented:
The implementation lgtm, I have added a test and made more pipes non pure in the PR linked above.
